### PR TITLE
docs: update the `auth.users` column names

### LINKED ISSUE #21545: Auth RLS docs inconsistent with `auth.users` schema
# Improve documentation

## Link 

[Using Row Level Security with Supabase Auth.](https://supabase.com/docs/guides/auth/row-level-security)

## Describe the problem

The page mentions that `auth.jwt()`:
> Returns the JWT of the user making the request. Anything that you store in the user's app_metadata column or the user_metadata column will be accessible using this function

... and yet the actual table columns do not feature such-named columns. Instead, I see `raw_app_meta_data` and `raw_user_meta_data` columns on the database for a project that was created a week ago

## Describe the improvement

Rename the columns to match the docs or fix the docs

## Additional context

See the screenshots below of my `auth.users` columns:
Screen Shot 2024-02-26 at 7 46 57 AM
Screen Shot 2024-02-26 at 7 46 48 AM

### PR DESCRIPTION
**Pull request by JBhrayn**
## What kind of change does this PR introduce?

docs update

## What is the current behavior?
The column names of `auth.users` mentioned on the docs, namely `app_metadata` and `user_metada` are outdated as shown in the following issue:

[#21545](https://github.com/supabase/supabase/issues/21545)

## What is the new behavior?
The said column names in docs now matches with the `auth.users` columns, namely `raw_app_meta_data` and `raw_user_meta_data`.

image

## Additional Context
Closes [#21545](https://github.com/supabase/supabase/issues/21545)

### CODE REVIEW

**Comment by charislam** [comment]
@J0, could you just give this a quick double-check on behalf of Team Auth? 🙇🏼

**Comment by J0** [comment]
Thanks for the tag @charislam 

Hey @JBhrayn,

Thanks for the PR! You are right that the columns on `auth.users are called `raw_app_metadata` and `raw_user_metadata`. The fields on the JWT are `app_metadata` and `user_metadata` though so you'd still call `auth.jwt() ->`app_metadata` to access the relevant fields.

Let me know if this hasn't been working for you though or feel free to add more context around the impetus for the rename.

Thanks!

File: apps/docs/content/guides/auth/row-level-security.mdx [hunk 1]
```diff
@@ -73,23 +73,23 @@ Returns the ID of the user making the request.
...
 
-- `user_metadata` - can be updated by the authenticated user using the `supabase.auth.update()` function. It is not a good place to store authorization data.
-- `app_metadata` - cannot be updated by the user, so it's a good place to store authorization data.
+- `raw_user_meta_data` - can be updated by the authenticated user using the `supabase.auth.update()` function. It is not a good place to store authorization data.
+- `raw_app_meta_data` - cannot be updated by the user, so it's a good place to store authorization data.
 
-The `auth.jwt()` function is extremely versatile. For example, if you store some team data inside `app_metadata`, you can use it to determine whether a particular user belongs to a team. For example, if this was an array of IDs:
+The `auth.jwt()` function is extremely versatile. For example, if you store some team data inside `raw_app_meta_data`, you can use it to determine whether a particular user belongs to a team. For example, if this was an array of IDs:
```
Comment (J0) [inline apps/docs/content/guides/auth/row-level-security.mdx:81]: Let's change this back to app_metadata

File: apps/docs/content/guides/auth/row-level-security.mdx [hunk 2]
```diff
@@ -73,23 +73,23 @@ Returns the ID of the user making the request.
...
 
 ```sql
 create policy "User is in team"
 on my_table
 to authenticated
-using ( team_id in (select auth.jwt() -> 'app_metadata' -> 'teams'));
+using ( team_id in (select auth.jwt() -> 'raw_app_meta_data' -> 'teams'));
```
Comment (J0) [inline apps/docs/content/guides/auth/row-level-security.mdx:87]: and thsi part too

**Comment by JBhrayn** [comment]
Thanks @charislam and @J0!

**Pull request closed without merging**